Signal/src/ViewControllers/ColorPickerViewController.swift

```diff
@@ -4,138 +4,184 @@
 
 import Foundation
 
-let colorSwatchHeight: CGFloat = 40
+protocol ColorViewDelegate: class {
+    func colorViewWasTapped(_ colorView: ColorView)
+}
 
 class ColorView: UIView {
-    let color: UIColor
-    let swatchView: UIView
+    public weak var delegate: ColorViewDelegate?
+    public let conversationColor: OWSConversationColor
+
+    private let swatchView: UIView
+    private let selectedRing: UIView
+    public var isSelected: Bool = false {
+        didSet {
+            self.selectedRing.isHidden = !isSelected
+        }
+    }
 
-    required init(color: UIColor) {
-        self.color = color
+    required init(conversationColor: OWSConversationColor) {
+        self.conversationColor = conversationColor
         self.swatchView = UIView()
+        self.selectedRing = UIView()
 
         super.init(frame: .zero)
+        self.addSubview(selectedRing)
+        self.addSubview(swatchView)
 
-        swatchView.backgroundColor = color
+        let cellHeight: CGFloat = 64
 
-        self.swatchView.layer.cornerRadius = colorSwatchHeight / 2
+        selectedRing.autoSetDimensions(to: CGSize(width: cellHeight, height: cellHeight))
+        selectedRing.layer.cornerRadius = cellHeight / 2
+        selectedRing.layer.borderColor = UIColor.ows_gray60.cgColor
+        selectedRing.layer.borderWidth = 2
+        selectedRing.autoPinEdgesToSuperviewEdges()
+        selectedRing.isHidden = true
 
-        self.addSubview(swatchView)
+        swatchView.backgroundColor = conversationColor.primaryColor
+        let swatchHeight: CGFloat = 48
```

Comment:
* Maybe `swatchSize` would be better?
* Does this assume that there's space for `kRowLength` swatches of size `swatchHeight` plus spacing, margins in all forms factors?

Reply:
> Maybe swatchSize would be better?

Sure, changed.

> Does this assume that there's space for kRowLength swatches of size swatchHeight plus spacing, margins in all forms factors?

Yup.